File: src/main/kotlin/Main2.java
import humc.lab.ext.core.proxy.ProxyFactoryByJDK;
import humc.lab.ext.demo.MyBizObj;
import humc.lab.ext.demo.NameSpi;
import humc.lab.ext.demo.NameSpiImpl1;
import humc.lab.ext.demo.NameSpiImpl2;


/**
 * @author: Mingchuan Hu(Kevin)
 * @date: 2023-11-20 22:31
 * @description
 */
public class Main2 {
    public static void main(String[] args) {
        new Main2().test2();
    }

    public void test2() {
        NameSpiImpl1 instance = NameSpiImpl1.INSTANCE;
        NameSpiImpl2 instance2 = NameSpiImpl2.INSTANCE;

        NameSpi spi = ProxyFactoryByJDK.Companion.proxy(NameSpi.class);
        MyBizObj bizObj = new MyBizObj("hi");

        spi.first(it -> {
            it.enrichName(bizObj);
            return null;
        });

        spi.enrichName(bizObj);
        System.out.println(bizObj.getName());
    }
}

File: src/main/kotlin/Main2.kt
import humc.lab.aef.core.ext.proxy.Level1ProxyByJDK
import com.example.demo.MyBizObj
import com.example.demo.NameSpi
import com.example.demo.scenario1.NameSpiImpl1
import com.example.demo.scenario2.NameSpiImpl2
import humc.lab.aef.core.session.BusinessSessionBuilder

fun main(args: Array<String>) {
    test3()
}

fun test3() {
    println("Hello World!")

    // Try adding program arguments via Run/Debug configuration.
    // Learn more about running applications: https://www.jetbrains.com/help/idea/running-applications.html.
    val spi: NameSpi = Level1ProxyByJDK.proxy(NameSpi::class)
    var bizObj = MyBizObj("hi")


    BusinessSessionBuilder
    spi.first().enrichName(bizObj)
    spi.enrichName(bizObj)
    println(bizObj.name)
}
